joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/querying/MiscTests.scala

```diff
@@ -258,4 +258,15 @@ class MiscTests extends RubyCode2CpgFixture {
         .size shouldBe 1
     }
   }
+
+  "CPG for code with singleton class" should {
```

Comment:
I'm afraid the support for `class << self` is not that straightforward, as it's typically used to build class methods. Please take a moment to explore alternatives to this representation and we can discuss their pros.

Reply:
Can you help me with 1-2 examples of where this would not work?

Reply:
Off the top of my head, it seems that an empty `class << self .. end` can just be disregarded, since its purpose is to add members to the outer class and there's nothing to add. When it's non-empty, I'm not sure, but probably its members should be moved to the outer class. But as as I said, this is me thinking out loud, so please do some investigation to understand this feature and to decide which representation is best.

Reply:
@xavierpinho can this be merged? at least this can fix few error handling. We will uncover more scenarios driven by testing, we can prioritise fixing more common issues first. Merging this will enable us to test those repos again for other issues.

cc: @tuxology

Reply:
@pandurangpatil I'd propose to merge just the null handling, then. (Going back to the AstCreator clean-up discussion.)

Reply:
@xavierpinho Rewriting `className` to return an `Option[String]` is one part of it. @rahul-privado will get the implementation updated.

Could you provide specific other improvements/handling that you want to change? Give specific suggestion, so that we can incorporate those suggestions and merge this PR with updated comments.

Reply:
@xavierpinho 
I have put in code to handle this better but not completely. This will avoid the crash though it is insufficient to make it work properly. Can we push this in so that it unblocks us?
I have added a disabled test case and a comment to put it in the radar. I will continue pursuing this.

Reply:
Thanks!

> Could you provide specific other improvements/handling that you want to change? Give specific suggestion, so that we can incorporate those suggestions and merge this PR with updated comments.

Right now I'd need to dive into this feature to provide concrete suggestions, besides the ones I already mentioned previously here. If @rahul-privado has a translation plan for this, let's hear it! Otherwise, I'd just say we don't handle this case besides the null handling (i.e. it'd be as if the `class << self ... end` portion didn't exist.)

Reply:
> @xavierpinho I have put in code to handle this better but not completely. This will avoid the crash though it is insufficient to make it work properly. Can we push this in so that it unblocks us? I have added a disabled test case and a comment to put it in the radar. I will continue pursuing this.

To be honest, I don't particularly like having a `self` class declared (cf. [test](https://github.com/joernio/joern/pull/3188/files#diff-8b764a636f592aa5d5298074cd5c1784e3553ba4ce4e5abff511cd4a1fd47cc5R268)). This isn't a class declaration and worse we can end up with many of those. If we don't have a general solution, I'd propose:

* `getName` returns None for `class << self`
* The `getName` caller interprets this `None` result and returns an empty `Seq[Ast]`

Reply:
Changed `className()` as per your suggestion. I do not have a concrete way to solve this problem right now and need more time to figure it out.